core/autoscaling.go

```diff
@@ -334,6 +335,40 @@ func (a *autoScalingGroup) replaceOnDemandInstanceWithSpot(
 	return a.detachAndTerminateOnDemandInstance(odInst.InstanceId)
 }
 
+func (a *autoScalingGroup) getInstanceTypeByTagInASG() (string, error) {
```

Comment:
You think? or it is ?  if it is, wish you would have caught this earlier, I spent sometime creating a test for this function.  Let me know.